chapter8.adoc

```diff
@@ -1,37 +1,146 @@
 [[chapter8]]
 [[Smsdedbg]]
-== `Smsdedbg`: Supervisor Domain External Debug
+== Supervisor Domain External Trace and Debug
 
-The <<MSDCFG>> CSR contains the `sdedbgalw` bit that controls whether the
-current scheduled SD is allowed to be external-debugged cite:[ExtDbg]. This bit
-is context switched (along with rest of the `msdcfg`) per SD.
+The <<MSDCFG>> CSR contains the `sdedbgalw` and `sdetrcalw` bits that control
+if the current scheduled SD is allowed to be external-debugged cite:[ExtDbg]
+or to be external-traced, respectively. External trace refers to tracing
+software running on a separate target RISC-V platform, via a trace control/data
+transport that provides external access to the trace encoder controls
+cite:[ETrc]. This extension only affects debug or trace orchestrated by an
+external actor. Self-hosted debug (and trace) are managed by the RDSM.
 
-=== `Smsdedbg` interaction with external debug security controls (Informative)
+=== `Smsdedbg`: External Debug allowed control for Supervisor Domain
+
+When M-mode external debug is enabled, all supervisor domains may also be
+debugged by an external debugger irrespective of the configuration held in
+`msdcfg.SDEDBGALW`.
+
+When M-mode external debug is disabled, whether execution at privilege modes
+less than `M-mode` may be debugged by an external debugger depends on the
+configuration held in `msdcfg.SDEDBGALW`.
+
+When `msdcfg.SDEDBGALW` = 0, external debug is disallowed for the SD. Abstract
+commands and halt request from the debug module are suppressed and stay pending
+while the SD is active, on a per-hart basis. Other debug operations (current or
+defined in the future) must similarly be kept pending while the SD is active on
+a hart.
+
+When `msdcfg.SDEDBGALW` = 1 then external debug of privilege modes less than
+`M-mode` is allowed for the SD on a per-hart basis, and:
+
+* A valid request e.g. `haltreq` may transition the hart to Debug Mode. See
+other cases in RISC-V Debug specification cite:[ExtDbg].
+* Abstract commands and program buffer execution can access state of privilege
+modes less than `M-mode`. In this context, "state" includes all resources
+accessible per the Debug specification cite:[ExtDbg].
```

Comment:
It is contradictory, debug spec states the it have "God" mode privilege and could access all resources, not constrained by privilege modes.

Reply:
sorry I don't understand what you are referring to as contradictory - can you clarify?
we are stating when M-mode external debug is disabled, whether execution at privilege modes less than `M-mode` may be debugged by an external debugger depends on the configuration held in `msdcfg.SDEDBGALW` -  so only when additionally `msdcfg.SDEDBGALW` = 1 then external debug of privilege modes less than `M-mode` is allowed for the SD on a per-hart basis, and within that scope all state of the SD can be accessed.

Reply:
I mean ""state" includes all resources accessible per the Debug specification", contradicts the "access state of privilege modes less than `M-mode`".
People get confused when read smmtt and debug spec

Reply:
will it help to restate as: 
"* Abstract commands and program buffer execution can access state of the SD that opted in to allow external debug. Here state includes system resources of privilege modes less than `M-mode` accessible per the Debug specification cite:[ExtDbg] ." 
- hope this clearly qualifies the statement.

Reply:
Actually the statement go against the debug specification. Instead of "per the Debug specification", it should be per risc-v privileged spec.

Reply:
I think the "all resources accessible" is the right term. There are 3 abstract commands - access register, quick access, and access memory. The access register command is used to access abstract registers (table 4) - CSRs, GPRs, and floating point registers. The quick access is used to execute the program buffer. Access memory is used to access memory. Program buffer execution can be used to execute instructions. The CSR, GPRs, and FP registers must be the state of the SD i.e. CSRs cannot be M-mode or D-mode CSRs. The instructions executed from program buffer cannot access any M-mode CSRs.

Reply:
In addition to CSRs, memory access should be protected by PMP, MTT. E.g. a SD should not be able to access other SD's memory or M-mode memory (if memory region is not shared). 
In summary, referring to ExtDbg spec seems the correct way to me.